util.go

```diff
@@ -318,3 +318,20 @@ func assertConfStatesEquivalent(l Logger, cs1, cs2 pb.ConfState) {
 	}
 	l.Panic(err)
 }
+
+// extend appends vals to the given dst slice. It differs from the standard
+// slice append only in the way it allocates memory. If cap(dst) is not enough
+// for appending the values, precisely size len(dst)+len(vals) is allocated.
+//
+// Use this instead of standard append in situations when this is the last
+// append to dst, so there is no sense in allocating more than needed.
+func extend[T any](dst, vals []T) []T {
```

Comment:
This will make raft/v3 incompatible with go 1.17 and under, should we avoid that by monomorphizing explicitly?

cc @ahrtr probably a general discussion to be had about which min Go version to target.

Reply:
Done.

Reply:
Good to see the change is reverted.